monkey/monkey_island/cc/services/reporting/report.py

```diff
@@ -545,28 +561,41 @@ def get_issues():
         logger.info("Issues generated for reporting")
         return issues_dict
 
-    @staticmethod
-    def is_latest_report_exists():
+    @classmethod
+    def get_latest_event_timestamp(cls) -> Optional[float]:
+        if not cls._agent_event_repository:
+            raise RuntimeError("Agent event repository does not exist")
+
+        # TODO: Add `get_latest_event` to the IAgentEventRepository
+        try:
+            agent_events = cls._agent_event_repository.get_events()
+            latest_timestamp = max(agent_events, key=lambda event: event.timestamp).timestamp
+        except (RetrievalError, ValueError) as err:
+            logger.error(f"Error occurred while getting latest timestamp from events: {err}")
+            return None
+
+        return latest_timestamp
+
+    @classmethod
+    def report_is_outdated(cls) -> bool:
         """
         This function checks if a monkey report was already generated and if it's the latest one.
-        :return: True if report is the latest one, False if there isn't a report or its not the
-        latest.
+        :return: True if the report is outdated, False if there is already a report or it is up to
+        date.
         """
-        latest_report_doc = mongo.db.report.find_one({}, {"meta_info.latest_monkey_modifytime": 1})
-
-        if latest_report_doc:
-            report_latest_modifytime = latest_report_doc["meta_info"]["latest_monkey_modifytime"]
-            latest_monkey_modifytime = Monkey.get_latest_modifytime()
-            return report_latest_modifytime == latest_monkey_modifytime
+        if cls._report:
```

Comment:
# If there are no agents, report should be empty
return not cls._agent_repository.get_agents()